### PR TITLE
Add test coverage for Registry

### PR DESCRIPTION
**Pull request by jeffbi**
Addresses parts of #4148 

Add coverage for `Get-Item`, `Get-ChildItem`, `Set-Item`, and `Clear-Item`

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
Missing tests:

- [ ] [invalid hive root](https://codecov.io/gh/PowerShell/PowerShell/src/master/src/System.Management.Automation/namespaces/RegistryProvider.cs#L174)
- [ ] validate `-whatif` works for `set-item` and `clear-item`
- [ ] attempt accessing path you don't have access to

**Comment by jeffbi** [comment]
@SteveL-MSFT These test run with the tag `RequireAdminOnWindows`. Is there a Registry path that admin would not have access to?

**Comment by SteveL-MSFT** [comment]
@jeffbi try `HKLM:\sam`

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 1]
```diff
@@ -125,6 +169,22 @@ Describe "Basic Registry Provider Tests" -Tags @("CI", "RequireAdminOnWindows")
             $property."$testPropertyName" | Should Be 0
         }
 
+        It "Verity Clear-Item" {
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Typo Verity->Verify

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]
Minor comment

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 2]
```diff
@@ -125,6 +169,22 @@ Describe "Basic Registry Provider Tests" -Tags @("CI", "RequireAdminOnWindows")
...
+            Set-ItemProperty -Path $testKey -Name $testPropertyName -Value $testPropertyValue
+            Set-Item -Path $testKey -Value $defaultPropertyValue
+            Clear-Item -Path $testKey
+            $key = Get-Item -Path $testKey
+            $key.Property.Length | Should BeExactly 0
+        }
+
+        It "Verity Clear-Item with -WhatIf" {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Typo here has well `Verify`

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by jeffbi** [comment]
Typos fixed.

The Travis failure seems to be in a test for Invoke-Item.

**Comment by SteveL-MSFT** [comment]
@jeffbi I have a PR to change the invoke-item tests to Feature as they shouldn't be CI anyways

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**